cmd/thermal-recorder/main.go

```diff
@@ -164,6 +182,10 @@ func handleConn(conn net.Conn, conf *Config, turret *TurretController, recording
 		rawFrame.ToFrame(frame)
 		totalFrames++
 
+		if snapshotPending.Load() == true {
+			snapshotChan <- frameToSnapshot(conf.OutputDir, frame)
+		}
```

Comment:
Instead of using the Value and channel, what about using the FrameLoop which was added recently? It provides access to all the recent frames and even has a `Previous` method which probably gives exactly what you want here. If the dbus service object had access to the FrameLoop then it could grab a frame from there whenever the call was made. This keeps the main thermal-recorder loop simpler.

If you went with this approach you'd have to add a mutex to protect the external methods of FrameLoop but this is easy to do. For extra safety you'd probably also want to make Previous return a copy instead of the frame directly from the FrameLoop.

Note that the one use of the FrameLoop.Previous call that currently exists goes away when we update thermal-recorder to use the latest go-cptv version. In fact, we should probably do that first before landing your snapshot work.

Overall this would be a clearer approach I think.

Reply:
OK, I'll update to use the FrameLoop when the other code has landed.